src/message/MessageContainer.js

```diff
@@ -1,27 +1,49 @@
 import React from 'react';
+import { Linking } from 'react-native';
 import htmlparser from 'htmlparser2';
 
 import renderHtmlChildren from './html/renderHtmlChildren';
 import { narrowFromMessage } from '../utils/narrow';
 import MessageFull from './MessageFull';
 import MessageBrief from './MessageBrief';
+import { isUrlInAppLink, getFullUrl, getMessageIdFromLink, getNarrowFromLink } from '../utils/url';
+import { getMessages } from '../api';
 
 const htmlToDomTree = html => {
   let domTree = null;
-  const parser = new htmlparser.Parser(new htmlparser.DomHandler((err, dom) => {
-    if (!err) domTree = dom;
-  }));
+  const parser = new htmlparser.Parser(
+    new htmlparser.DomHandler((err, dom) => {
+      if (!err) domTree = dom;
+    }),
+  );
   parser.write(html);
   parser.done();
   return domTree;
 };
 
 export default class MessageContainer extends React.PureComponent {
-
   handlePress = () => {
     const { message, doNarrow } = this.props;
     doNarrow(narrowFromMessage(message), message.id);
-  }
+  };
+
+  inAppLinkPress = async href => {
+    const { users, auth, doNarrow } = this.props;
+    const anchor = getMessageIdFromLink(href, auth.realm);
+    const narrow = getNarrowFromLink(href, auth.realm, users);
+    let message;
+    if (anchor) {
+      message = await getMessages(auth, anchor, 0, 0, narrow);
```

Comment:
makes sense 👍  We could also alter the current doNarrow implementation to use an anchor in that as well.